google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/ErrorCountPerConnectionTest.java

```diff
@@ -162,52 +226,48 @@ public void readOverTwoPeriods() throws Exception {
         // noop
       }
     }
-    ArgumentCaptor<Long> errorCountCaptor = ArgumentCaptor.forClass(long.class);
-    Mockito.doNothing()
-        .when(statsRecorderWrapperForConnection)
-        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
-    runInterceptorTasksAndAssertCount();
-    List<Long> allErrorCounts = errorCountCaptor.getAllValues();
-    assertThat(allErrorCounts.size()).isEqualTo(1);
-    assertThat(allErrorCounts.get(0)).isEqualTo(errorCount);
 
-    errorCount = 0;
+    runInterceptorTasksAndAssertCount();
+    long errorCount2 = 0;
 
     for (int i = 0; i < 20; i++) {
       Query query;
       if (i % 3 == 0) {
         query = Query.create(SUCCESS_TABLE_NAME);
       } else {
         query = Query.create(ERROR_TABLE_NAME);
-        errorCount += 1;
+        errorCount2 += 1;
       }
       try {
         stub.readRowsCallable().call(query).iterator().hasNext();
       } catch (Exception e) {
         // noop
       }
     }
-    errorCountCaptor = ArgumentCaptor.forClass(long.class);
-    Mockito.doNothing()
-        .when(statsRecorderWrapperForConnection)
-        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
+
     runInterceptorTasksAndAssertCount();
-    allErrorCounts = errorCountCaptor.getAllValues();
-    assertThat(allErrorCounts.size()).isEqualTo(1);
-    assertThat(allErrorCounts.get(0)).isEqualTo(errorCount);
-  }
 
-  @Test
-  public void ignoreInactiveConnection() throws Exception {
-    EnhancedBigtableStub stub = EnhancedBigtableStub.create(builder.build());
+    Collection<MetricData> allMetrics = metricReader.collectAllMetrics();
```

Comment:
so the points recorded using `perConnectionErrorCountHistogram.record()` stay in the metricReader until `metricReader.collectAllMetrics()` is called and they're picked up? 
Since we're using a 60-second cycle for both collecting and exporting the metrics, I wonder if it's possible that on some minutes we could skip exporting metrics and end up sending them twice over the next minute. For example, we start the collection at time 59.998s and finish at 60.001s, but export at 60.000s. Next minute, we start the export at 199.998s until 199.999s and export at 120.000s, double-exporting the metrics. (I guess if it is indeed an issue, it would've also been the case before migrating to otel?)

Reply:
The StatsWrapper was introduced because we want a fixed OpenCensus version and namespace, and it's not a required abstraction once we migrated to OTEL. 

In this test, we use a in-memory metric reader which let us investigate the actual aggregated values so we don't need to use a mock anymore. We still trigger the task run in `runInterceptorTasksAndAssertCount` method. The issue you described could be a problem, but yes, I think it would exist before the migration.